### PR TITLE
Add Serial Config message and more protocol

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

File: uavcan/tunnel/Protocol.uavcan [hunk 1]
```diff
@@ -4,5 +4,8 @@
 #
 
 uint8 MAVLINK                   = 0     # MAVLink
+uint8 MAVLINK2                  = 1     # MAVLink2
+uint8 GPS                       = 2     # GPS
```
Comment (julianoes) [inline uavcan/tunnel/Protocol.uavcan:8]: Wouldn't the protocol be NMEA or UBX or something like that?

File: uavcan/tunnel/2011.SerialConfig.uavcan [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#
+# This message carries baudrate for tunnel channel_id
+
+uint8 channel_id
```
Comment (julianoes) [inline uavcan/tunnel/2011.SerialConfig.uavcan:4]: I'm not sure how this works or what this means. Some description might be helpful.

File: uavcan/tunnel/Protocol.uavcan (hunk 1 above)
Comment (bugobliterator) [reply uavcan/tunnel/Protocol.uavcan:8]: Ardupilot autodetects the specific protocol. I guess a better definition would be `GPS_GENERIC`

File: uavcan/tunnel/2011.SerialConfig.uavcan [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#
+# This message carries baudrate for tunnel channel_id
```
Comment (tridge) [inline uavcan/tunnel/2011.SerialConfig.uavcan:2]: worth adding options to allow for inversion, flow control etc. we have room in an 8 byte packet, so 16 bit options would be worth having for future expansion

**Review by tridge — CHANGES_REQUESTED** [review]

File: uavcan/tunnel/2011.SerialConfig.uavcan [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+# The channelID allows for additional routing between the source and target nodes.
+uint8 channel_id
+
+uint32 baud
+
+uint32 options
```
Comment (julianoes) [inline uavcan/tunnel/2011.SerialConfig.uavcan:9]: How are the options encoded? Or would this be additional enums?

File: uavcan/tunnel/Protocol.uavcan [hunk 5]
```diff
@@ -4,5 +4,8 @@
 #
 
 uint8 MAVLINK                   = 0     # MAVLink
+uint8 MAVLINK2                  = 1     # MAVLink2
+uint8 GPS_GENERIC               = 2     # GPS
```
Comment (julianoes) [inline uavcan/tunnel/Protocol.uavcan:8]: What about?

```suggestion
uint8 GPS_GENERIC               = 2     # GPS in any protocol (e.g. UBX, NMEA, etc.)
```

File: uavcan/tunnel/2011.SerialConfig.uavcan (hunk 2 above)
Comment (julianoes) [reply uavcan/tunnel/2011.SerialConfig.uavcan:5]: It's still not clear to me what this means or how it works, but I'm probably missing UAVCAN context.

File: uavcan/tunnel/2011.SerialConfig.uavcan (hunk 4 above)
Comment (tridge) [inline uavcan/tunnel/2011.SerialConfig.uavcan:9]: lets make options 24 bit so this fits in a single bxCAN packet

**Review by tridge — CHANGES_REQUESTED** [review]

File: uavcan/tunnel/2011.SerialConfig.uavcan (hunk 4 above)
Comment (bugobliterator) [reply uavcan/tunnel/2011.SerialConfig.uavcan:9]: additional enums will be required, I am letting them not defined until there is actual implementation.

**Review by tridge — APPROVED** [review]

**Review by julianoes — APPROVED** [review]

**Pull request closed without merging**